pkg/activator/handler/handler.go

```diff
@@ -61,7 +69,53 @@ func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	err := a.Throttler.Try(revID, func() {
-		attempts, httpStatus := a.proxyRequest(w, r, target)
+		var (
+			attempts   int
+			httpStatus int
+		)
+		// If a GET probe interval has been configured, then probe
+		// the queue-proxy with our network probe header until it
+		// returns a 200 status code.
+		success := (a.GetProbeCount == 0)
+		if !success {
+			probeReq := &http.Request{
+				Method:     http.MethodGet,
+				URL:        target,
+				Proto:      r.Proto,
+				ProtoMajor: r.ProtoMajor,
+				ProtoMinor: r.ProtoMinor,
+				Host:       r.Host,
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(network.ProbeHeaderName): []string{"true"},
+				},
+			}
+			settings := wait.Backoff{
+				Duration: 100 * time.Millisecond,
+				Factor:   1.3,
+				Steps:    a.GetProbeCount,
+			}
+			err := wait.ExponentialBackoff(settings, func() (bool, error) {
+				probeResp, err := a.Transport.RoundTrip(probeReq)
+				if err != nil {
+					a.Logger.Errorw("Pod probe failed", zap.Error(err))
+					return false, nil
+				}
+				if probeResp.StatusCode != http.StatusOK {
```

Comment:
Do we want to do similar to the retry roundtripper where we only continue on 503? I worry that retrying on any !ok response will surface e.g. 500 with a 1min delay on activation rather than instant fail.

Reply:
I changed this to only continue on 503, and below to only declare success on `http.StatusOK`.